modules/carousel/Carousel.tsx

```diff
@@ -222,79 +248,106 @@ const Carousel: React.FC<ICarouselProps & WithStylesProps<Theme, ICarouselStyles
     addEvent("mouseleave", scrollableElementRef?.current, () =>
       onMouseUp(scrollableElementRef?.current)
     );
-  }, [scrollableElementRef]);
+  }, [blockScroll, scrollableElementRef]);
 
   const calculateChildrenSize = useCallback<() => number>(() => {
     if (!scrollableElementRef?.current) {
       return 0;
     }
 
-    const element: HTMLElement = scrollableElementRef.current;
+    const element = scrollableElementRef.current.children[scrollType === "centered" ? 1 : 0];
 
     return (
-      (element.children[0]?.clientWidth || 0) +
+      (element?.clientWidth || 0) +
       (borderWidth ||
-        Number(
-          (element.children[0] as HTMLElement)?.style?.borderLeftWidth?.replace("px", "") || 0
-        )) +
+        Number((element as HTMLElement)?.style?.borderLeftWidth?.replace("px", "") || 0)) +
       (borderWidth ||
-        Number(
-          (element.children[0] as HTMLElement)?.style?.borderRightWidth?.replace("px", "") || 0
-        )) +
+        Number((element as HTMLElement)?.style?.borderRightWidth?.replace("px", "") || 0)) +
       marginBetweenItems
     );
-  }, [borderWidth, marginBetweenItems, scrollableElementRef]);
+  }, [borderWidth, marginBetweenItems, scrollType, scrollableElementRef]);
 
   const scroll = useCallback(
     (rightDir = true) => {
       if (scrollableElementRef && scrollableElementRef.current) {
         const element = scrollableElementRef.current;
-        const currentScroll = element.scrollLeft ?? 0;
-        const scrollElementWidth = element.clientWidth || 0;
-        const halfMarginBetweenItems = marginBetweenItems / 2;
+        if (scrollType === "normal") {
```

Comment:
Maybe we could set the scrollTypes to constants?

Reply:
I don't see any advantage on that since creating an enum will require it to be bundled, when with this it won't. 
what do you think?